test/e2e/node/pod_resize.go

```diff
@@ -538,6 +538,214 @@ func doPodResizeRetryDeferredTests(f *framework.Framework) {
+
+		testPod4CPUQuantityResized := resource.NewMilliQuantity(nodeAvailableMilliCPU/2+nodeAvailableMilliCPU/8, resource.DecimalSI)
+		testPod4MemoryQuantityResized := initialMemoryQuantity
+
+		patchTestpod2ToDeferred := fmt.Sprintf(`{
+			"spec": {
+				"containers": [
+					{
+						"name":      "c",
+						"resources": {
+							"requests": {
+								"cpu": "%dm",
+								"memory": "%d"
+							},
+							"limits": {
+								"cpu": "%dm"
+							}
+						}
+					}
+				]
+			}
+		}`, testPod2CPUQuantityResized.MilliValue(), testPod2MemoryQuantityResized.Value(), testPod2CPUQuantityResized.MilliValue())
+
+		patchTestpod3ToDeferred = fmt.Sprintf(`{
+			"spec": {
+				"containers": [
+					{
+						"name":      "c",
+						"resources": {
+							"requests": {
+								"cpu": "%dm",
+								"memory": "%d"
+							},
+							"limits": {
+								"cpu": "%dm"
+							}
+						}
+					}
+				]
+			}
+		}`, testPod3CPUQuantityResized.MilliValue(), testPod3MemoryQuantityResized.Value(), testPod3CPUQuantityResized.MilliValue())
+
+		patchTestpod4ToDeferred := fmt.Sprintf(`{
+			"spec": {
+				"containers": [
+					{
+						"name":      "c",
+						"resources": {
+							"requests": {
+								"cpu": "%dm",
+								"memory": "%d"
+							},
+							"limits": {
+								"cpu": "%dm"
+							}
+						}
+					}
+				]
+			}
+		}`, testPod4CPUQuantityResized.MilliValue(), testPod4MemoryQuantityResized.Value(), testPod4CPUQuantityResized.MilliValue())
+
+		patches := []string{patchTestpod2ToDeferred, patchTestpod3ToDeferred, patchTestpod4ToDeferred}
+
+		for i := range patches {
+			testPod := testPods[i+1]
+			patch := patches[i]
+			ginkgo.By(fmt.Sprintf("Resize pod '%s' that cannot fit node due to insufficient CPU or memory", testPod.Name))
+			testPod, err = f.ClientSet.CoreV1().Pods(testPod.Namespace).Patch(ctx,
+				testPod.Name, types.StrategicMergePatchType, []byte(patch), metav1.PatchOptions{}, "resize")
+			framework.ExpectNoError(err, "failed to patch pod for resize")
+			framework.ExpectNoError(e2epod.WaitForPodCondition(ctx, f.ClientSet, testPod.Namespace, testPod.Name, "display pod resize status as deferred", f.Timeouts.PodStart, func(pod *v1.Pod) (bool, error) {
+				return helpers.IsPodResizeDeferred(pod), nil
+			}))
+			gomega.Expect(testPod.Generation).To(gomega.BeEquivalentTo(2))
+		}
+
+		ginkgo.By("deleting pod 1")
+		delErr1 = e2epod.DeletePodWithWait(ctx, f.ClientSet, testPod1)
+		framework.ExpectNoError(delErr1, "failed to delete pod %s", testPod1.Name)
+
+		ginkgo.By(fmt.Sprintf("Verify pod '%s' is resized successfully after pod deletion '%s'", testPod2.Name, testPod1.Name))
+		expected = []podresize.ResizableContainerInfo{
+			{
+				Name: "c",
+				Resources: &cgroups.ContainerResources{
+					CPUReq: testPod2CPUQuantityResized.String(),
+					CPULim: testPod2CPUQuantityResized.String(),
+					// Convert to KiB because ExpectPodResized does a string comparison.
```

Comment:
which comparison is that?

Reply:
need to find it again (will do so tomorrow), but upon reflection it will probably be better to just update it so that it compares by value rather than string

Reply:
don't have time rn to look into why the test helper thinks `31574112256` != `30834094Ki`, will try to look into it later. For now, updating this test with tweaked numbers